streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -54,15 +58,17 @@
     private final Map<String, Set<TaskId>> consumerToPreviousStatefulTaskIds = new TreeMap<>();
 
     // the following four maps are used only for logging purposes;
-    // TODO KAFKA-10283: we could consider merging them with other book-keeping maps at client-levels
-    //                   so that they would not be inconsistent
-    private final Map<String, Set<TaskId>> consumerToPreviousActiveTaskIds = new TreeMap<>();
-    private final Map<String, Set<TaskId>> consumerToAssignedActiveTaskIds = new TreeMap<>();
-    private final Map<String, Set<TaskId>> consumerToAssignedStandbyTaskIds = new TreeMap<>();
-    private final Map<String, Set<TaskId>> consumerToRevokingActiveTaskIds = new TreeMap<>();
-
+    private final Map<String, Map<TaskId, Set<ConsumerState>>> consumerToTaskIdsConsumerStates = new TreeMap<>();
```

Comment:
This map is great as a concise & efficient way to keep track of the tasks, but I'm worried that we're sacrificing some readability. I'm not sure we necessarily want to stuff everything into a single data structure, just reduce the duplication of effort in tracking tasks at both the client and consumer level. If you take a look at the `activeTasks` family of sets above, that's basically just a superset of all tasks in the `consumerToAssignedActive` map.

But `consumerToAssignedActive` and `consumerToAssignedStandby`, for example, actually track two completely orthogonal things, so I think it seems reasonable for them to be in two different data structures. WDYT?